jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java

```diff
@@ -111,29 +122,30 @@ public void acceptSearchMatch(SearchMatch match) {
                     try {
                         if (method.isMainMethod()) {
                             IResource resource = method.getResource();
-                            if (resource != null) {
-                                IProject project = resource.getProject();
-                                if (project != null) {
-                                    String mainClass = method.getDeclaringType().getFullyQualifiedName();
-                                    IJavaProject javaProject = getJavaProject(project);
-                                    if (javaProject != null) {
-                                        String moduleName = getModuleName(javaProject);
-                                        if (moduleName != null) {
-                                            mainClass = moduleName + "/" + mainClass;
-                                        }
-                                    }
-                                    String projectName = ProjectsManager.DEFAULT_PROJECT_NAME.equals(project.getName()) ? null : project.getName();
-                                    String filePath = null;
-                                    if (match.getResource() instanceof IFile) {
-                                        try {
-                                            filePath = match.getResource().getLocation().toOSString();
-                                        } catch (Exception ex) {
-                                                // ignore
-                                        }
-                                    }
-                                    res.add(new PackageNode(mainClass, filePath, NodeKind.FILE));
+                            if (resource == null) {
+                                return;
+                            }
+                            IProject project = resource.getProject();
```

Comment:
Is `project` been used here?

Reply:
> Is `project` been used here?

It is used for getting module name previously. I will remove it.